src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -455,7 +456,6 @@ private SolutionState AddProject(ProjectId projectId, ProjectState projectState)
             // changed project list so, increment version.
             var newSolutionAttributes = _solutionAttributes.With(version: Version.GetNewerVersion());
 
-            var newProjectIds = ProjectIds.ToImmutableArray().Add(projectId);
```

Comment:
general idea is that you don't need to compute this if you're already updating the projectIdMap.  it's effectively inferred by the fact that you passed in something different.